FloatingBaseEstimatorDevice faills if some dependancies are not installed

In https://github.com/robotology/robotology-superbuild/pull/526 I'm trying to port blf in the superbuild. We decided to add the possibility to install `blf` without having `manif`, `qhull`, `cppad` and `casadi`. If those dependencies are not installed the compilation fails with this error 
 
This is the [workflow](https://github.com/robotology/robotology-superbuild/pull/526/checks?check_run_id=1449165575)
```
2020-11-24T23:17:54.8097418Z -- Created target RobotInterface for export BipedalLocomotionFramework.
2020-11-24T23:17:54.8098937Z --  [ ] Plugin: FloatingBaseEstimatorDevice (ENABLE_FloatingBaseEstimatorDevice)
2020-11-24T23:17:54.8100330Z -- Created device FloatingBaseEstimatorDevice.
2020-11-24T23:17:54.8101330Z -- Generating the Urdf Model from generate_stairs_urdf
2020-11-24T23:17:54.8102038Z -- Configuring done
2020-11-24T23:17:54.8103382Z CMake Error at /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/share/yarp/cmake/YarpPlugin.cmake:584 (add_library):
2020-11-24T23:17:54.8104761Z   Target "FloatingBaseEstimatorDevice" links to target
2020-11-24T23:17:54.8105842Z   "BipedalLocomotion::FloatingBaseEstimators" but the target was not found.
2020-11-24T23:17:54.8187370Z   Perhaps a find_package() call is missing for an IMPORTED target, or an
2020-11-24T23:17:54.8188061Z   ALIAS target is missing?
2020-11-24T23:17:54.8188560Z Call Stack (most recent call first):
2020-11-24T23:17:54.8189519Z   cmake/AddBipedalLocomotionYARPDevice.cmake:36 (yarp_add_plugin)
2020-11-24T23:17:54.8190823Z   devices/FloatingBaseEstimatorDevice/CMakeLists.txt:7 (add_bipedal_yarp_device)
2020-11-24T23:17:54.8191528Z 
2020-11-24T23:17:54.8191750Z 
```

cc @prashanthr05

--- a/devices/FloatingBaseEstimatorDevice/CMakeLists.txt
+++ b/devices/FloatingBaseEstimatorDevice/CMakeLists.txt
@@ -2,14 +2,13 @@
 # This software may be modified and distributed under the terms of the
 # GNU Lesser General Public License v2.1 or any later version.
 
-if(FRAMEWORK_COMPILE_YarpImplementation)
+if(FRAMEWORK_COMPILE_YarpImplementation AND FRAMEWORK_COMPILE_FloatingBaseEstimators)
 
-add_bipedal_yarp_device(
-  NAME FloatingBaseEstimatorDevice
-  TYPE BipedalLocomotion::FloatingBaseEstimatorDevice
-  SOURCES src/FloatingBaseEstimatorDevice.cpp
-  PUBLIC_HEADERS include/BipedalLocomotion/FloatingBaseEstimatorDevice.h
-  PUBLIC_LINK_LIBRARIES ${YARP_LIBRARIES} ${iDynTree_LIBRARIES} BipedalLocomotion::YarpUtilities BipedalLocomotion::RobotInterfaceYarpImplementation BipedalLocomotion::ParametersHandlerYarpImplementation BipedalLocomotion::FloatingBaseEstimators
-  INI FloatingBaseEstimatorDevice.ini)
+  add_bipedal_yarp_device(
+    NAME FloatingBaseEstimatorDevice
+    TYPE BipedalLocomotion::FloatingBaseEstimatorDevice
+    SOURCES src/FloatingBaseEstimatorDevice.cpp
+    PUBLIC_HEADERS include/BipedalLocomotion/FloatingBaseEstimatorDevice.h
+    PUBLIC_LINK_LIBRARIES ${YARP_LIBRARIES} ${iDynTree_LIBRARIES} BipedalLocomotion::YarpUtilities BipedalLocomotion::RobotInterfaceYarpImplementation BipedalLocomotion::ParametersHandlerYarpImplementation BipedalLocomotion::FloatingBaseEstimators
+    INI FloatingBaseEstimatorDevice.ini)
 endif()
-

